lib/puppet/pops/types/types.rb

```diff
@@ -695,7 +695,7 @@ def self.register_ptype(loader, ir)
   end
 
   def instance?(o, guard = nil)
-    if o.is_a?(String) || o.is_a?(Numeric) || o.is_a?(TrueClass) || o.is_a?(FalseClass) || o.is_a?(Regexp)
+    if o.instance_of?(String) || o.instance_of?(Numeric) || o.instance_of?(TrueClass) || o.instance_of?(FalseClass) || o.instance_of?(Regexp)
```

Comment:
Puppet 6.x supports ruby 2.3 and up, and in ruby 2.3 `Numeric` is the base class for several different number related classes:

```
Numeric
 |- Integer
 |    |- Fixnum
 |    |- Bignum
 |
 |- Float
```

In ruby 2.4, `Bignum` and `Fixnum` were unified: https://rubyreferences.github.io/rubychanges/2.4.html#fixnum-and-bignum-are-unified-into-integer

So we could keep `is_a?(Numeric)` (and add a note calling out the special case) or test each case. I'd lean towards the first option.